scripts/cat/save_load.py

```diff
@@ -0,0 +1,151 @@
+        inter_cat = cat_class.all_cats[cat]
+
+        # Add ID to list of faded cats.
+        faded_ids.append(cat)
+
+        # If they have a mate, break it up
+        if inter_cat.mate:
+            for mate_id in inter_cat.mate:
+                if mate_id in cat_class.all_cats:
+                    cat_class.all_cats[mate_id].unset_mate(inter_cat)
+
+        # If they have parents, add them to their parents "faded offspring" list:
+        for x in inter_cat.get_parents():
+            if x in cat_class.all_cats:
+                cat_class.all_cats[x].faded_offspring.append(cat)
+            else:
+                parent_faded = add_faded_offspring_to_faded_cat(clanname, x, cat)
+                if not parent_faded:
+                    print(f"WARNING: Can't find parent {x} of {cat.name}")
+
+        # Get a copy of info
+        if get_game_setting("save_faded_copy"):
+            copy_of_info += (
+                ujson.dumps(inter_cat.get_save_dict(), indent=4)
+                + "\n--------------------------------------------------------------------------\n"
+            )
+
+        # SAVE TO ITS OWN LITTLE FILE. This is a trimmed-down version for relation keeping only.
+        cat_data = inter_cat.get_save_dict(faded=True)
+
+        safe_save(f"{get_save_dir()}/{clanname}/faded_cats/{cat}.json", cat_data)
+
+        # Remove the cat from the active cats lists
+        game.clan.remove_cat(
+            cat
+        )  # todo: when catdirectory is added, this dependency injection can be removed
+
+    cat_to_fade = []
+
+    # Save the copies, flush the file.
+    if get_game_setting("save_faded_copy"):
+        with open(
+            get_save_dir() + "/" + clanname + "/faded_cats_info_copy.txt",
+            "a",
+            encoding="utf-8",
+        ) as write_file:
+            if not os.path.exists(
+                get_save_dir() + "/" + clanname + "/faded_cats_info_copy.txt"
+            ):
+                # Create the file if it doesn't exist
+                with open(
+                    get_save_dir() + "/" + clanname + "/faded_cats_info_copy.txt",
+                    "w",
+                    encoding="utf-8",
+                ) as create_file:
+                    pass
+
+            with open(
+                get_save_dir() + "/" + clanname + "/faded_cats_info_copy.txt",
+                "a",
+                encoding="utf-8",
+            ) as write_file:
+                write_file.write(copy_of_info)
+
+                write_file.flush()
+                os.fsync(write_file.fileno())
+
+
+def add_faded_offspring_to_faded_cat(clanname, parent: str, offspring: str):
+    """In order to siblings to work correctly, and not to lose relation info on fading, we have to keep track of
+    both active and faded cat's faded offpsring. This will add a faded offspring to a faded parents file.
+    """
+    try:
+        with open(
+            get_save_dir() + "/" + clanname + "/faded_cats/" + parent + ".json",
```

Comment:
Moving forward, we should try to use Pathlib instead of constructing path literals as strings.